scrapy/core/scraper.py

```diff
@@ -182,7 +184,7 @@ def _process_spidermw_output(self, output, request, response, spider):
         """
         if isinstance(output, Request):
             self.crawler.engine.crawl(request=output, spider=spider)
-        elif isinstance(output, (BaseItem, dict)):
+        elif isinstance(output, (BaseItem, dict)) or is_dataclass_instance(output):
```

Comment:
Maybe we should implement a high-level helper function for use cases like this?
```suggestion
        elif is_item(output):
```

This would make our current FAQ entry about splitting an item into multiple items much more readable.